text/0000-buildpack-registry-search-api.md

```diff
@@ -0,0 +1,235 @@
+        }
+      }
+    }
+  ]
+  ```
+  
+- **GET /buildpacks/:ns/:name**
+
+  Retrieves metadata for a specific buildpack.
+  ```
+  $ curl https://registry.buildpacks.io/v1/buildpacks/projectriff/command-function
+  ```
+  ```json
+  {
+    "latest": {
+      "description": "The Command Function Buildpack is a Cloud Native Buildpack V3 that provides riff Command Function Invoker to functions",
+      "license": "MIT",
+      "ns":"projectriff",
+      "name":"command-function",
+      "version": "1.4.1",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/command-function@sha256:99f9054abb73635a9b251b61d3627a8ff86508c767f9d691c426d45e8758596f"  
+    },
+    "versions": {
+      "1.4.1": {
+        "link": "https://registry.buildpacks.io/buildpacks/v1/projectriff/command-function/1.4.1"
+      },
+      "1.3.9": {
+        "link": "https://registry.buildpacks.io/buildpacks/v1/projectriff/command-function/1.3.9"
+      }
+    }
+  } 
+
+- **GET /buildpacks/:ns/:name/:version**
+
+  Retrieves metadata for a specific buildpack version (`:version` must be a semver or `latest`).  This response *may* contain more metadata e.g. download metrics.  Since `description` and `license` can change between different versions, it should therefore ONLY be included for each specific buildpack version.
+  ```
+  $ curl https://registry.buildpacks.io/buildpacks/v1/projectriff/command-function/1.4.1
+  ```
+  ```json
+  {
+      "description": "The Command Function Buildpack is a Cloud Native Buildpack V3 that provides riff Command Function Invoker to functions",
+      "license": "MIT",
+      "ns":"projectriff",
+      "name":"command-function",
+      "version": "1.4.1",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/command-function@sha256:99f9054abb73635a9b251b61d3627a8ff86508c767f9d691c426d45e8758596f"
+  }
+  ```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+In the initial implementation, we can take advantage of the existing [registry](https://github.com/buildpacks/registry-index) repository to pull buildpack data from.  To keep ensure up-to-date buildpack data, we'd have a polling loop that performs a `git pull`.  This polling would take place in a separate background process.  
+
+Each time the local repository has been updated via `git pull`, buildpack data will be processed/normalized into a single JSON object, where plain text searches can be used against it.   Fields in this JSON object will be re-indexed as searchable fields, which will be compared against the search text during the retrieval algorithm.
+
+Search fields will include `ns`, and `name`. 
+
+In addition, a server process will expose endpoints and handle incoming request for the GET endpoints mentioned earlier.  For search requests, the text will be extracted from the `query` query parameter and used to search against the normalized buildpacks index.   Results will be added to the server response, as a list of JSON objects.
```

Comment:
```suggestion
In addition, a server process will expose endpoints and handle incoming request for the GET endpoints mentioned earlier.  For search requests, the text will be extracted from the `matches` query parameter and used to search against the normalized buildpacks index.   Results will be added to the server response, as a list of JSON objects.
```
We renamed the query param to `matches` in the example but the text still says `query` in a couple places. I assume we want to change the text?